tcp_app_lib/tcp_server_app/tcp_server_app.cpp

```diff
@@ -0,0 +1,31 @@
+//
+// Created by Shubham Sawant on 08/07/21.
+//
+
+#include "tcp_server_app.h"
+
+TCPServerApp::TCPServerApp(std::string name, unsigned int port):
+      name_(name) {
+      TCPMessage::Handler app_message_handler =
+        [this](std::shared_ptr<TCPMessage> message) -> tcp_util::ACTION_ON_CONNECTION {
+        return this->HandleMessage(message);
+      };
+      server_ = std::make_shared<TCPServer>(
+                    port, app_message_handler,
+  GetMessageBufferCapacity());
+}
+
+TCPServerApp::~TCPServerApp() {} // Keep compiler happy :)
+
+tcp_util::ACTION_ON_CONNECTION
+TCPServerApp::HandleMessage(std::shared_ptr<TCPMessage> tcp_message) {
+    PRINT_THREAD_ID
+    std::cout << name_ << ":: Received message: " << tcp_message->message()->data()
```

Comment:
Since this is a multi threaded program, PRINT_THREAD_ID won;t work as intended. And actual log messages may get interleaved. 

Consider following approach: 
PRINT_THREAD_ID  <<  " :: Received message: " < ....; 

Even better:
Consider using https://github.com/google/glog which takes care of these things for you and also adds additional meta data like line numner, function name, trace_id (if available)